src/compiler/frontend.rs

```diff
@@ -731,6 +731,19 @@ fn frontend_start(
     }
 }
 
+/// Entrypoint for compilation.  This yields a CompileForm which represents a full
+/// program.
+///
+/// Given a CompilerOpts specifying the global options for the compilation, return
+/// a representation of the parsed program.  Desugaring is not done in this step
+/// so this is a close representation of the user's input, containing location
+/// references etc.
+///
+/// pre_forms is a list of forms, because most SExp readers, include parse_sexp
+/// parse a list of complete forms from a source text.  It is possible for frontend
+/// to use a list of forms, but it is most often used with a single list in
+/// chialisp.  Usually pre_forms will contain a slice containing one list or
```

Comment:
```suggestion
/// to use a list of forms, but it is most often used with a single list in
/// chialisp.  Usually pre_forms will contain a slice containing one list or
```

Describe how that "single list" is different than using a list of forms.